File: 08. Kadane's Algorithm/kadane.py
from sys import maxsize

size = int(input("Enter the nunmber of elements: \n"))

listOfElementsA = list(map(int,input("Enter the elements separated by space: \n").strip().split()))

max_so_far = -maxsize - 1
max_ending_here = 0

for i in range(0, size):
    max_ending_here = max_ending_here + listOfElementsA[i]
    if (max_so_far < max_ending_here):
        max_so_far = max_ending_here

    if max_ending_here < 0:
        max_ending_here = 0


print(max_so_far)


    

File: 08. Kadane's Algorithm/kadane.java
import java.io.*;
import java.util.Arrays;

class kadane
{
    public static void main(String args[])throws IOException
    {
        BufferedReader br =new BufferedReader(new InputStreamReader(System.in));
        System.out.println("Enter the number of Elements in array A:");
        int sizeA = Integer.parseInt(br.readLine());
        int listOfElementsA[] = new int[sizeA];
        System.out.println("Enter the Elements for A:");     
        int max_so_far = Integer.MIN_VALUE, max_ending_here = 0;
        String[] strs = br.readLine().trim().split("\\s+");
      
        for(int i=0;i<sizeA;i++)
        {
            listOfElementsA[i] = Integer.parseInt(strs[i]);
        }

        for (int i = 0; i < sizeA; i++)
        {
            max_ending_here = max_ending_here + listOfElementsA[i];
            if (max_so_far < max_ending_here)
                max_so_far = max_ending_here;
            if (max_ending_here < 0)
                max_ending_here = 0;
        }
        
        System.out.println(max_so_far);


    }
}